src/testIntegration/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactResourceUpdateContactTest.kt

```diff
@@ -0,0 +1,150 @@
+package uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.contact
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.springframework.http.MediaType
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.IntegrationTest
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.AuthenticationError
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.MalformedRequest
+import uk.gov.justice.digital.hmpps.sendlegalmailtoprisonsapi.config.NotFound
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class ContactResourceUpdateContactTest : IntegrationTest() {
+
+  private companion object {
+    val JOHN_SMITH = ContactRequest(
+      prisonerName = "John Smith",
+      prisonId = "BXI",
+      prisonNumber = "A1234BC"
+    )
+  }
+
+  @Test
+  fun `unauthorised without a valid auth token`() {
+    webTestClient.put()
+      .uri("/contact/1")
+      .accept(MediaType.APPLICATION_JSON)
+      .contentType(MediaType.APPLICATION_JSON)
+      .exchange()
+      .expectStatus().isUnauthorized
+  }
+
+  @Test
+  fun `forbidden without a valid role`() {
+    webTestClient.put()
+      .uri("/contact/1")
+      .accept(MediaType.APPLICATION_JSON)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(setAuthorisation(user = "AUSER_GEN"))
+      .bodyValue(JOHN_SMITH)
+      .exchange()
+      .expectStatus().isForbidden
+      .expectBody().jsonPath("$.errorCode.code").isEqualTo(AuthenticationError.code)
+  }
+
+  @Test
+  fun `bad request given empty body`() {
+    webTestClient.put()
+      .uri("/contact/1")
+      .accept(MediaType.APPLICATION_JSON)
+      .contentType(MediaType.APPLICATION_JSON)
+      .headers(setCreateBarcodeAuthorisation())
+      .bodyValue("{}")
+      .exchange()
+      .expectStatus().isBadRequest
+      .expectBody().jsonPath("$.errorCode.code").isEqualTo(MalformedRequest.code)
+  }
+
+  @Test
+  fun `not found if no id`() {
+    webTestClient.put()
+      .uri("/contact")
```

Comment:
Ah, this is the reason we've added the exception handler for method not supported -> not found
I'm not sure about that / this test .... 🤔 

I would argue that this test (`POST /contact`) is not calling the update contact method at all. Its calling the /contact method with a HTTP method that we don't support. I think letting spring return this as a 405 is the right approach here.

Worth a chat ?